service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/create/CreateControlledResourceFlight.java

```diff
@@ -102,6 +102,10 @@ public CreateControlledResourceFlight(FlightMap inputParameters, Object beanBag)
         throw new IllegalStateException(
             String.format("Unrecognized resource type %s", resource.getResourceType()));
     }
+    // Update private_resource_state from INITIALIZING to ACTIVE, if this is a private resource.
```

Comment:
It won't get cleaned up, it'll just be stuck in the WSM database. Since the whole creation process (no DB row -> INITIALIZING state -> ACTIVE state) is inside one Stairway flight, that can only happen with a dismal flight failure, but if it does we'll have to clean up manually.